src/WebJobs.Script/Scale/HostPerformanceManager.cs

```diff
@@ -99,5 +229,38 @@ internal ApplicationPerformanceCounters GetPerformanceCounters(ILogger logger =
 
             return null;
         }
+
+        private IFunctionInvocationDispatcher GetDispatcherAsync()
+        {
+            var hostManager = _serviceProvider.GetService<IScriptHostManager>();
+            var dispatcherFactory = (hostManager as IServiceProvider)?.GetService<IFunctionInvocationDispatcherFactory>();
+            if (dispatcherFactory != null)
+            {
+                var dispatcher = dispatcherFactory.GetFunctionDispatcher();
+                if (dispatcher.State == FunctionInvocationDispatcherState.Initialized)
```

Comment:
@yojagad - should this check change to !FunctionInvocationDispatcherState.Default as well?

Reply:
I think for sampling latency, I'd rather avoid pinging a worker that is in the process of starting up. I think it's best to wait until it's fully initialized. Also, the primary check we're doing below relies on getting cpu metrics from the workers, and we won't have those anyways if the worker hasn't been initialized.